src/fs/v1/PrivateTree.ts

```diff
@@ -0,0 +1,93 @@
+import { Tree, File, SemVer, HeaderV1 } from '../types'
+import { CID, FileContent } from '../../ipfs'
+import keystore from '../../keystore'
+import PublicTree  from './PublicTree'
+import { constructors as PrivateFileConstructors } from './PrivateFile'
+import header from './header'
+import semver from '../semver'
+
+
+export class PrivateTree extends PublicTree {
```

Comment:
I'm not super happy with how boilerplatey a lot of this class is (also not sure happy with having 2-3 "constructors" at the bottom of each Tree/File class). But I'm not really sure a way around it. let me know if you have any ideas!